File: src/led.cc
#include "led.h"

#include "gpio.h"

Led::Led() {}

void Led::setIntensity(double intensity) {
  intensity_ = intensity;
  int value = (int) (GPIO_LED_RANGE * intensity);
  Gpio::pwmWrite(GPIO_LED, value);
}

double Led::getIntensity() {
  return intensity_;
}

File: node/led.js
const gpio = require('./gpio');

let v;

function setIntensity(intensity) {
  v = Math.max(Math.min(1, intensity), 0);
  const valueInRange = Math.floor(gpio.GPIO_LED_RANGE * (1 - v));
  gpio.pwmWrite(gpio.GPIO_LED, valueInRange);
}

function getIntensity() {
  return v;
}

module.exports = { setIntensity, getIntensity };
